src/httpClient.ts

```diff
@@ -117,21 +123,27 @@ class DefaultHTTPClient implements HTTPClient {
     }
 
     const headers: Record<string, string> = {
-      // eslint-disable-next-line @typescript-eslint/naming-convention
-      "Accept-Encoding": "gzip",
       // eslint-disable-next-line @typescript-eslint/naming-convention
       Authorization: `Bearer ${this.apiKey}`,
       // eslint-disable-next-line @typescript-eslint/naming-convention
       "User-Agent": this.userAgent,
     };
+
+    if (this.compression) {
+      headers["Accept-Encoding"] = "gzip";
+    }
+
     if (body) {
       headers["Content-Type"] = "application/json";
     }
 
+    let requestCompressionDuration;
     let requestBody: Uint8Array | string | null = null;
-    if (body && compress) {
+    if (body && compress && this.compression) {
       headers["Content-Encoding"] = "gzip";
+      const beforeRequestCmpression = performance.now();
       requestBody = await gzipAsync(JSON.stringify(body));
+      requestCompressionDuration = performance.now() - beforeRequestCmpression;
```

Comment:
typo

Reply:
Oh, didn't realize this wasn't exercised in test, will write a test

Reply:
it is excercised in test, the typo is in the variable declaration and use

Reply:
rather than a new test just add an assertion to this list

https://github.com/turbopuffer/turbopuffer-typescript/blob/2b2b65938233c359c4033d21962567d78c1d214d/src/turbopuffer.test.ts#L175-L182

Reply:
I added it there, and I added some compression tests anyway to ensure all the paths are taken (including request compression)

Reply:
Oh, it was a consistent typo, not a functional one, yeah ok fixed